define/types.go

```diff
@@ -170,17 +175,30 @@ func TempDirForURL(dir, prefix, url string) (name string, subdir string, err err
 	return "", "", errors.Errorf("unreachable code reached")
 }
 
-func cloneToDirectory(url, dir string) ([]byte, error) {
-	gitBranch := strings.Split(url, "#")
+func cloneToDirectory(url, dir string) ([]byte, string, error) {
+	gitSubdir := ""
+	gitBranch := ""
+	// Check if git url also contains branch with subdir.
+	gitBranchPart := strings.Split(url, "#")
+	if len(gitBranchPart) > 1 {
+		gitSubDirPart := strings.Split(gitBranchPart[1], ":")
+		if len(gitSubDirPart) > 1 {
+			gitSubdir = gitSubDirPart[1]
+			gitBranch = gitSubDirPart[0]
+		} else {
+			gitBranch = gitSubDirPart[0]
+		}
+	}
 	var cmd *exec.Cmd
-	if len(gitBranch) < 2 {
+	if len(gitBranchPart) < 2 {
```

Comment:
You just did this check above, Combine this section with the if len(gitBranchPart) > 1 above

Reply:
Yes moved it below.